src/components/AppAlert/AppAlert.tsx

```diff
@@ -15,6 +15,21 @@ export const AppAlert = () => {
     alertContext.setAlert({ ...alertContext.alert, showAlert: false });
   };
 
+  useEffect(() => {
+    if (alertContext.alert.showAlert) {
```

Comment:
I suggest to replace the home made alert component with https://www.npmjs.com/package/react-alert
may you try with this PR ? (it has a timeout property)